core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -326,6 +327,18 @@ ServiceMap lookupDependenciesOf(ServiceMap resolved, Class<? extends Service> re
       for (Class<? extends Service> dependency : identifyImmediateDependenciesOf(requested)) {
         resolved = lookupService(resolved, dependency);
       }
+      for (Class<? extends Service> dependency : identifyImmediateOptionalDependenciesOf(requested)) {
+        try {
+          resolved = lookupService(resolved, dependency);
+        } catch (DependencyException de) {
```

Comment:
My idea was to move the try/catch inside identifyImmediateOptionalDependenciesOf

Reply:
That can't be done with the way the code currently is structured as you can only know if an optional dependency can be resolved or not when it is being looked up.

Reply:
Ah... Not that useful then